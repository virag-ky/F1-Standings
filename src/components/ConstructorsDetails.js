import React from 'react';
import { useSelector } from 'react-redux';
import Constructor from './Constructor';
import '../styles/constructors.css';

const ConstructorsDetails = () => {
  const constructors = useSelector((state) => state.constructor);

  return (
    <div className="constructorsDetailsContainer">
      <div className="constructorsStandings">
        <div className="constructorImg">
          <div className="mask" />
        </div>
        <div className="constructorsTitleDiv">
          <h2 className="constructorsTitle">CONSTRUCTOR&apos;S STANDINGS</h2>
        </div>
      </div>
      <table>
        <thead>
          <tr id="head">
            <th>NAME</th>
            <th>POSITION</th>
            <th>WINS</th>
            <th>POINTS</th>
          </tr>
        </thead>
        <tbody>
          {constructors.length > 0 ? (
            Object.values(constructors).map((constructor) => (
              <Constructor
                key={`number${constructor.position}`}
                name={constructor.name}
                position={constructor.position}
                wins={constructor.wins}
                points={constructor.points}
              />
            ))
          ) : (
            <tr id="noConstructors">
              <td>
                <h3>No constructors</h3>
              </td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

export default ConstructorsDetails;
