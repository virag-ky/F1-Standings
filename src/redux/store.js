import { combineReducers } from 'redux';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import homePageReducer, { fetchSeason } from './homepage/homepage';
import driversReducer, { fetchDrivers } from './drivers/drivers';
import constructorReducer, {
  fetchConstructors,
} from './constructors/constructors';
import racesReducer, { fetchRaces } from './races/races';

const currentYear = new Date().getFullYear().toString();

const reducers = combineReducers({
  home: homePageReducer,
  driver: driversReducer,
  constructor: constructorReducer,
  race: racesReducer,
});

const store = configureStore({
  reducer: reducers,
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
});
store.dispatch(fetchSeason(currentYear));
store.dispatch(fetchDrivers(currentYear));
store.dispatch(fetchConstructors(currentYear));
store.dispatch(fetchRaces(currentYear));

export default store;
