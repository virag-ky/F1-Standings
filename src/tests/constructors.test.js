import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import ConstructorsDetails from '../components/ConstructorsDetails';

describe("constructor's page renders correctly", () => {
  test("displays the title 'Constructor's Standings' ", () => {
    const { getByText } = render(
      <Provider store={store}>
        <ConstructorsDetails />
      </Provider>,
    );
    expect(getByText("CONSTRUCTOR'S STANDINGS")).toBeInTheDocument();
  });

  test('renders the page', () => {
    const constructorsPage = render(
      <Provider store={store}>
        <ConstructorsDetails />
      </Provider>,
    );
    expect(constructorsPage).toMatchSnapshot();
  });
});
