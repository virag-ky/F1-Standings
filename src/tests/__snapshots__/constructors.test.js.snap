// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constructor's page renders correctly renders the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="constructorsDetailsContainer"
      >
        <div
          class="constructorsStandings"
        >
          <div
            class="constructorImg"
          >
            <div
              class="mask"
            />
          </div>
          <div
            class="constructorsTitleDiv"
          >
            <h2
              class="constructorsTitle"
            >
              CONSTRUCTOR'S STANDINGS
            </h2>
          </div>
        </div>
        <table>
          <thead>
            <tr
              id="head"
            >
              <th>
                NAME
              </th>
              <th>
                POSITION
              </th>
              <th>
                WINS
              </th>
              <th>
                POINTS
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="constructorsDetailsContainer"
    >
      <div
        class="constructorsStandings"
      >
        <div
          class="constructorImg"
        >
          <div
            class="mask"
          />
        </div>
        <div
          class="constructorsTitleDiv"
        >
          <h2
            class="constructorsTitle"
          >
            CONSTRUCTOR'S STANDINGS
          </h2>
        </div>
      </div>
      <table>
        <thead>
          <tr
            id="head"
          >
            <th>
              NAME
            </th>
            <th>
              POSITION
            </th>
            <th>
              WINS
            </th>
            <th>
              POINTS
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
